### DISCUSSION TITLE
Is it possible to use a template in a run_once* file name?

### DISCUSSION THREAD

**Discussion by 0xjams** [twpayne/chezmoi#3775] [Q&A] [answered]
I don't know if I'm overcomplicating this, but I want to bootstrap my environment in either Windows or Linux. 
I have this file `run_once_install_ansible.sh.tmpl`. However, that file will not run as Powershell, so I was wondering if something like this is possible: `run_once_install_ansible{{ if eq .chezmoi.os "windows" }}.ps1{{ else }}.sh{{ end }}.tmpl`

Another option I was considering is to use .chezmoiignore to have a .sh and a PowerShell template for each OS so that only the run_once* file that corresponds to the OS gets copied. But I'm not sure if .chezmoiignore works for run* files.

What's the commonly used approach?

**Comment by Lockszmith-GH** [comment, marked as answer]
Did a quick search for `chezmoiignore` and `chezmoiscripts` and found #2510 

I then tested it like this in PowerShell:

```console
> Set-Location "$(chezmoi source-path)"
### Create the two flavors of run_once_script that will alternate based on os
> mkdir .chezmoiscripts
> '"Hello World! PS1"' > .chezmoiscripts/run_once_script.ps1
> '#!/bin/bash'+"`n"+'echo "Hello World! sh"' > .chezmoiscripts/run_once_script.sh
### test it out - what's managed, what's ignored
> chezmoi ignored

> chezmoi managed
.chezmoiscripts/script.ps1
.chezmoiscripts/script.sh

### add template pattern to .chezmoiignore
> '.chezmoiscripts/script{{ if eq .chezmoi.os "windows" }}.sh{{ else }}.ps1{{ end }}' >> .chezmoiignore
### Test the tempalte syntax (on windows - the .sh should be ignored)
> Get-Content .chezmoiignore | chezmoi.exe execute-template
.chezmoiscripts/script.sh

### test again, this time with the .chezmoiignore entry in place
> chezmoi ignored
.chezmoiscripts/script.sh

> chezmoi managed
.chezmoiscripts/script.ps1
```

I'm guessing this is what you were asking for. Good Luck

**Comment by twpayne** [comment]
>  I was wondering if something like this is possible: run_once_install_ansible{{ if eq .chezmoi.os "windows" }}.ps1{{ else }}.sh{{ end }}.tmpl

No, it's not currently possible to templatize filenames. Use `.chezmoiignore` as @Lockszmith-GH says.
